File: app/src/main/java/com/wagdybuild/retrofitproject/data/PostClients.java
package com.wagdybuild.retrofitproject.data;

import com.wagdybuild.retrofitproject.models.Comment;
import com.wagdybuild.retrofitproject.models.Comment2;
import com.wagdybuild.retrofitproject.models.Post;
import com.wagdybuild.retrofitproject.models.Post2;

import java.util.ArrayList;
import retrofit2.Call;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class PostClients {
    //private static final String BASE_URL = "https://jsonplaceholder.typicode.com/";
    private static final String BASE_URL = "https://jsonplaceholder.org/";
    private final PostInterface postInterface;
    private static PostClients INSTANCE;

    public PostClients() {
        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl(BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .build();

        postInterface = retrofit.create(PostInterface.class);

    }

    public static PostClients getINSTANCE() {
        if(INSTANCE == null){
            INSTANCE = new PostClients();
        }
        return INSTANCE;
    }

    public Call<ArrayList<Post2>> getPosts(){
        return postInterface.getPosts();
    }
    public Call<ArrayList<Comment2>> getComments(String postId){
        return postInterface.getComments(postId);
    }
}

File: app/src/main/java/com/wagdybuild/retrofitproject/data/PostClients.kt
package com.wagdybuild.retrofitproject.data

import com.wagdybuild.retrofitproject.models.Comment2
import com.wagdybuild.retrofitproject.models.Post2
import retrofit2.Call
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.create

class PostClients {
    companion object {
        //private static final String BASE_URL = "https://jsonplaceholder.typicode.com/";
        private const val BASE_URL = "https://jsonplaceholder.org/"
    }
    private val api by lazy {
        Retrofit.Builder()
            .baseUrl(BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .build().create(PostInterface::class.java)
    }
    suspend fun getPosts(): List<Post2> = api.getPosts()
    suspend fun getComments(postId: String):List<Comment2> = api.getComments(postId)


}